streams/src/main/java/org/apache/kafka/streams/AutoOffsetReset.java

```diff
@@ -0,0 +1,76 @@
+package org.apache.kafka.streams;
+
+import java.time.Duration;
+import java.util.Optional;
+
+/**
+ * Sets the {@code auto.offset.reset} configuration when
+ * {@link #addSource(AutoOffsetReset, String, String...) adding a source processor} or when creating {@link KStream}
+ * or {@link KTable} via {@link StreamsBuilder}.
+ */
+public class AutoOffsetReset {
+    protected final Optional<Long> duration;
+
+    protected AutoOffsetReset(Optional<Long> duration) {
+        this.duration = duration;
+    }
+
+    /**
+     * Creates an AutoOffsetReset instance representing "latest".
+     * 
+     * @return an AutoOffsetReset instance for the "latest" offset.
+     */
+    public static AutoOffsetReset latest() {
+        return new AutoOffsetReset(Optional.empty());
+    }
+
+    /**
+     * Creates an AutoOffsetReset instance representing "earliest".
+     * 
+     * @return an AutoOffsetReset instance for the "earliest" offset.
+     */
+    public static AutoOffsetReset earliest() {
+        return new AutoOffsetReset(Optional.of(0L));
+    }
+
+    /**
+     * Creates an AutoOffsetReset instance with a custom duration.
+     * 
+     * @param duration the duration to use for the offset reset; must be non-negative.
+     * @return an AutoOffsetReset instance with the specified duration.
+     * @throws IllegalArgumentException if the duration is negative.
+     */
+    public static AutoOffsetReset duration(Duration duration) {
+        if (duration.isNegative()) {
+            throw new IllegalArgumentException("Duration cannot be negative");
+        }
+        return new AutoOffsetReset(Optional.of(duration.toMillis()));
+    }
+
+    /**
+     * Retrieves the offset reset duration if specified.
+     * 
+     * @return an Optional containing the duration in milliseconds, or empty if "latest".
+     */
+    public Optional<Long> getDuration() {
```

Comment:
This method should not be on this class, but we would add an `AutoOffsetResetInternal` class inside the `internal` package and add it there (cf the KIP discussion (or `GroupedInternal` which extends `Grouped` as an example: https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/kstream/internals/GroupedInternal.java)

Reply:
Would you please explain why this is the case? What is the reasoning behind not wanting this on the class? I am asking because I am not sure I understand why, and would learn from your take.

I will add a class based on the example you gave, thank you for taking the time to find an example.

Reply:
The reason is, that a user of Kafka Streams, never needs to call this method. This method is basically useless to the user, and we try to have a minimal API that we provide to the user, to avoid any distraction/noise in the API.

Thus, we "hide" this method that we only need internally, by only adding it to an internal sub-class.

Does this make sense?

Reply:
I see, we may have a need to do this internally but a user never would, and giving them the ability to call it would not make their lives any easier but just adds bloat. I appreciate the explanation, I am beginning to see the relationship between the 'internals' and the user-facing components and will keep this in mind in the future. I have some things to learn yet about the API we are exposing to the end-user